http-client/Network/HTTP/Client/Types.hs

```diff
@@ -657,9 +662,11 @@ instance Show Request where
         , "}"
         ]
 
-redactSensitiveHeader :: Header -> Header
-redactSensitiveHeader ("Authorization", _) = ("Authorization", "<REDACTED>")
-redactSensitiveHeader h = h
+redactSensitiveHeader :: [HeaderName] -> Header -> Header
+redactSensitiveHeader _ ("Authorization", _) = ("Authorization", "<REDACTED>")
```

Comment:
I'm not sure I agree with having `Authorization` sit separately from the rest of these. I'd lean towards having the default `redactHeaders` be a singleton set of just `"Authorization"`